src/Compiler/Microsoft.AspNetCore.Razor.Language/test/IntegrationTests/ComponentCodeGenerationTestBase.cs

```diff
@@ -10478,7 +10522,27 @@ @using Microsoft.AspNetCore.Components.Web
 
         // Assert
         AssertDocumentNodeMatchesBaseline(generated.CodeDocument);
-        AssertCSharpDocumentMatchesBaseline(generated.CodeDocument, verifyLinePragmas: false);
+        AssertCSharpDocumentMatchesBaseline(generated.CodeDocument);
+        var result = CompileToAssembly(generated, throwOnFailure: false);
+        if (DesignTime)
+        {
+            result.Diagnostics.Verify(
+                // x:\dir\subdir\Test\TestComponent.cshtml(2,54): error CS0103: The name 'start' does not exist in the current context
```

Comment:
> Should there be an RZ diagnostic here too? Or is that verified elsewhere?

Good catch, thanks. Looks like it's missing for this test case.

> I'm also curious how this looks in the IDE.

I'm also curious, but couldn't make local razor work with VS lately (17.8 nor 17.9) - many cryptic error boxes appear; maybe I will try resetting the VS hives.

Reply:
> Should there be an RZ diagnostic here too?

Actually there is a RZ diagnostic about the mixed content, it just hadn't changed in this PR so it's not visible in the diff. Anyway, will assert this.

Reply:
> I suspect we (tooling) need to filter out these C# diagnostics that overlap with the broader mixed content RZ diagnostic.

I fixed `@formname` codegen so these errors went away. But I added new test demonstrating superfluous C# errors when there's mixed content - that's because the compiler emits `TypeCheck(x y z)` - it could perhaps emit `TypeCheck(x + y + z)` instead or just `TypeCheck(x); TypeCheck(y); TypeCheck(z);` - I do the latter for `@formname` but seems like a larger refactoring would be needed to do it for component attributes as well.

Reply:
> I'm also curious how this looks in the IDE.

I'm able to run VS now. Added some screenshots to PR description.

Reply:
Thanks for the screenshots, I think it looks good. Some people might complain if they see lots and lots of diagnostics in a single tooltip, but I think the C# diagnostic actually helps understand what is going on - the Razor compiler is clearly expecting a string in you example.